### PR TITLE
Add `status_of_primary` and `status_of_secondary` to `azure_storage_account` table

### PR DESCRIPTION
**Pull request by pdecat**
Add `status_of_primary` and `status_of_secondary` to `azure_storage_account` table

### CODE REVIEW

File: azure/table_azure_storage_account.go [hunk 1]
```diff
@@ -342,6 +348,12 @@ func tableAzureStorageAccount(_ context.Context) *plugin.Table {
 				Type:        proto.ColumnType_STRING,
 				Transform:   transform.FromField("Account.AccountProperties.SecondaryLocation"),
 			},
+			{
+				Name:        "status_of_secondary",
+				Description: "The status indicating whether the secondary location of the storage account is available or unavailable. Only available if the SKU name is Standard_GRS or Standard_RAGRS. Possible values include: 'Available', 'Unavailable'",
```
Comment (misraved) [inline azure/table_azure_storage_account.go:353]:
```suggestion
				Description: "The status indicating whether the secondary location of the storage account is available or unavailable. Only available if the SKU name is Standard_GRS or Standard_RAGRS. Possible values include: 'Available', 'Unavailable'.",
```

**Review by misraved — COMMENTED** [review]
Thanks @pdecat for adding the new columns 👍. Could you please add minor example queries in the table docs reflecting the usage of the new columns?

File: azure/table_azure_compute_virtual_machine.go [hunk 2]
```diff
@@ -61,6 +61,12 @@ func tableAzureComputeVirtualMachine(_ context.Context) *plugin.Table {
 				Description: "The type of the resource in Azure.",
 				Type:        proto.ColumnType_STRING,
 			},
+			{
+				Name:        "creation_date",
+				Description: "The creation date of the vm.",
```
[the remark this answers is not in the document]
Comment (pdecat) [reply azure/table_azure_compute_virtual_machine.go:66]: Hi @cbruno10, I usually refer to the SDK, but you are right, I've fooled myself on this one as:
* I copied it from another resource as we had it in our original Microsoft Graph (non Steampipe) queries.
* it is not available in the currently used version of the SDK: https://github.com/Azure/azure-sdk-for-go/blob/v58.3.0/services/compute/mgmt/2020-06-01/compute/models.go#L11676
* it is only available in newer SDK versions but as `TimeCreated`: https://github.com/Azure/azure-sdk-for-go/blob/v68.0.0/services/compute/mgmt/2022-08-01/compute/models.go#L18388

Marked this PR as draft for now, sorry for the noise.

[the remark this answers is not in the document]
Comment (pdecat) [reply azure/table_azure_compute_virtual_machine.go:66]: Also, I've just checked our CMDB which was not fed with Steampipe, and all of our 6k+ VMs have their creation time field empty...

[the remark this answers is not in the document]
Comment (pdecat) [reply azure/table_azure_compute_virtual_machine.go]: I've dropped this from the PR.

File: azure/table_azure_storage_account.go [hunk 3]
```diff
@@ -342,6 +348,12 @@ func tableAzureStorageAccount(_ context.Context) *plugin.Table {
 				Type:        proto.ColumnType_STRING,
 				Transform:   transform.FromField("Account.AccountProperties.SecondaryLocation"),
 			},
+			{
+				Name:        "status_of_secondary",
+				Description: "The status indicating whether the secondary location of the storage account is available or unavailable. Only available if the SKU name is Standard_GRS or Standard_RAGRS. Possible values include: 'available', 'unavailable'.",
```
Comment (pdecat) [inline azure/table_azure_storage_account.go:353]: Go SDK states `Available` and `Unavailable` with capital letters, but actual values are lowercase.

https://github.com/Azure/azure-sdk-for-go/blob/v58.3.0/services/storage/mgmt/2019-06-01/storage/models.go#L522

**Review by misraved — APPROVED** [review]

**Pull request merged**